src/window.cpp

```diff
@@ -64,26 +64,17 @@ bool Window::visible() { return glfwGetWindowAttrib(handle, GLFW_FOCUSED); }
 Window::Window() {
     glfwSetErrorCallback(glfw_error_callback);
     if (!glfwInit()) log_fatal("Failed to initialise GLFW");
-#if __APPLE__
-    glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
-    glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 2);
-    glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);  // 3.2+ only
-    glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);            // Required on Mac
-    glsl_version = "#version 150";
-    log_info("Set up on Apple: OpenGL 3.2 Core forward compatible GLSL v150");
-#else
-    glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
-    glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 0);
-    //glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);  // 3.2+ only
-    //glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);            // 3.0+ only
-    glsl_version = "#version 130";
-    log_info("Set up on Linux: OpenGL 3.0 GLSL v130");
-#endif
-    // glEnable(GL_MULTISAMPLE);
-    // glfwWindowHint(GLFW_SAMPLES, 4);
-    // Create window with graphics context
+    log_info("Setting up window with OpenGL 4.1 Core forward compatible GLSL v410");
```

Comment:
Hurray!

Reply:
> Looks good. Builds and works without setting the MESA env var. No more segfault loading the nml.
> 
> However, loading the `morph.nml` in question shows no fustrums. I do see a few points. Under regions there are exclamation marks next to all regions (dend,axon,soma). E.g. `dend`: `(join (join (join (join (join nil (join nil (segment 3))) (join nil (segment 4))) (join nil (segment 9))) (join nil (segment 10))) (join nil (segment 5)))` and the error suggests two candidates for `join()`. I see some chatter in the Slack so maybe you've seen it already.

Yes, that's a bug in Arbor, to be fixed there and separately.